src/components/DataVisualization/Table/Table.vue

```diff
@@ -130,6 +157,106 @@ export default {
 
+    onWindowResize(e) {
+      this.setWindowWidth(e);
+      this.setHiddenColumns();
+    },
+
+    setHiddenColumns() {
+      this.state.hiddenColumnsIndex = [];
+      this.$nextTick(() => {
+        const columnsOffset = this.headerColumns
+          .map(node => node.offsetTop)
+          .filter(value => value !== undefined);
+        if (!columnsOffset.length) return;
+
+        /**
+         * Here, we are verifying if the column header was wrapped.
+         * If yes, ew need to hide this column.
+         */
+        this.state.hiddenColumnsIndex = columnsOffset.reduce((acc, curr, index) => {
+          if (index === 0) return acc;
+
+          if (curr === columnsOffset[0]) return acc;
+
+          if (this.hasActionColumn && index === this.header.length)
+            acc.push('action');
+          else acc.push(index);
+
+          return acc;
+        }, []);
+
+        /**
+         * If there is any hidden column, we need to verify if there is enough space to the expand icon appears.
+         * If not, we need to subtract the last column size.
+         * If yes, we need to verify what size the expand icon should be.
+         */
+
+        if (this.state.hiddenColumnsIndex) {
+          // eslint-disable-next-line no-nested-ternary
+          const sizeWord = this.windowWidth < 576 ? '' : this.windowWidth < 768 ? 'sm-' : this.windowWidth < 993 ? 'md-' : 'lg-';
+          const sizeClass = `pb-col-${sizeWord}`;
+
+          const headerColumnSize = this.shownHeadersColumns.reduce((acc, curr, index) => {
+            /** expand div */
+            if (index === this.shownHeadersColumns.length - 1) return acc;
+
+            const classList = Array.from(curr.classList);
+            const currentClass = classList.find(value => value.includes(sizeClass));
+            return [...acc, Number(currentClass?.replace(/\D/g, ''))];
+          }, []);
+
+          const totalSize = headerColumnSize.reduce((acc, curr) => acc + curr, 0);
+          const shouldSubtractLastColumnSize = totalSize >= 12;
+          this.state.shouldSubtractLastColumnSize = shouldSubtractLastColumnSize;
+          this.state.columnWithReducedSize = shouldSubtractLastColumnSize
+            ? headerColumnSize.length - 1
+            : null;
+          this.state.expandRowsColumnSize = Math.max(12 - totalSize, 1);
+        } else {
+          this.state.columnWithReducedSize = null;
+        }
+      });
+    },
+
+    /**
+     * This function is responsible for add the classes to the columns;
+     */
+    columnClasses(size, index) {
+      /**
+       * We are using the variable shouldSubtractLastColumnSize to verify if we need to subtract the last column size.
+       */
+      const shownHeaders = Array.from(Array.from(
+        document.body.getElementsByClassName('table-header-container')[0]
+          ?.childNodes || [],
+      ).filter(value => value.offsetTop !== undefined))
+        .filter((_, idx) => !this.state.hiddenColumnsIndex.includes(idx));
```

Comment:
n tem uma maneira mais clean de fazer isso ? separar esses caras em uma função talvez

Reply:
I split into 2 variables to be more clear